### PR TITLE
Refactor CDCPD_Moveit_Node

### PR DESCRIPTION
**Pull request by dcolli23**
Fairly major refactor where I:
- gutted much of the CDCPD_Moveit_Node constructor
- Added a CDCPD_Moveit_Node header file
- Added both CDCPD_Moveit_Node and CDCPD parameter structs.
- took out all lambda functions of the CDPCD_Moveit_Node constructor
  and made them methods of the CDCPD_Moveit_Node class
- added a DeformableObjectConfiguration class for initializing and
  keeping track of the deformable object tracking
  - These changes will facilitate the choice between tracking ropes vs
    cloths at runtime instead of having to hardcode the specification/
    maintain a different branch for the two.

Next steps are to refactor the CDCPD class and refactor gripper information from the CDCPD_Moveit_Node

### CODE REVIEW

File: cdcpd/include/cdcpd/cdcpd.h [hunk 1]
```diff
@@ -68,6 +68,25 @@ enum OutputStatus {
   Success,
 };
 
+class CdcpdParameters
```
Comment (PeterMitrano) [inline cdcpd/include/cdcpd/cdcpd.h:71]: Might be nice to keep all caps, to be consistent with CDCPD class.

**Comment by PeterMitrano** [comment]
Overall a good idea! I'm not reviewing closely but proceed anyways.

File: cdcpd/include/cdcpd/cdcpd.h (hunk 1 above)
Comment (dcolli23) [reply cdcpd/include/cdcpd/cdcpd.h]: Got it. Addressed in the most recent commit

**Comment by dcolli23** [comment]
Thanks for the comments! They're addressed so I'll squash and merge

**Pull request merged**